src/Uno.UWP/ApplicationModel/Calls/PhoneCallHistoryManager.Android.cs

```diff
@@ -0,0 +1,145 @@
+			}
+
+			if(!Windows.Extensions.PermissionsHelper.IsDeclaredInManifest(Android.Manifest.Permission.ReadCallLog))
+				return null;
+
+			// required for contact name
+			if (!Windows.Extensions.PermissionsHelper.IsDeclaredInManifest(Android.Manifest.Permission.ReadContacts))
+				return null;
+
+			if (accessType == PhoneCallHistoryStoreAccessType.AllEntriesReadWrite)
+			{
+				if (!Windows.Extensions.PermissionsHelper.IsDeclaredInManifest(Android.Manifest.Permission.WriteCallLog))
+					return null;
+			}
+
+			List<string> requestPermission = new List<string>();
+
+			// check what permission should be granted
+			if (! await Windows.Extensions.PermissionsHelper.CheckPermission(CancellationToken.None, Android.Manifest.Permission.ReadCallLog))
+			{
+				requestPermission.Add(Android.Manifest.Permission.ReadCallLog);
+			}
+
+			if (! await Windows.Extensions.PermissionsHelper.CheckPermission(CancellationToken.None, Android.Manifest.Permission.ReadContacts))
+			{
+				requestPermission.Add(Android.Manifest.Permission.ReadContacts);
+			}
+
+			if (accessType == PhoneCallHistoryStoreAccessType.AllEntriesReadWrite)
+			{
+				if (!await Windows.Extensions.PermissionsHelper.CheckPermission(CancellationToken.None, Android.Manifest.Permission.WriteCallLog))
+				{
+					requestPermission.Add(Android.Manifest.Permission.WriteCallLog);
+				}
+			}
+
+			if (requestPermission.Count < 1)
+				return _histStore;
+
+			// in my tests, this ends with ERROR!! about null object reference inside TryGetPermission, on second call
+			foreach (var sPerm in requestPermission)
+				await Windows.Extensions.PermissionsHelper.TryGetPermission(CancellationToken.None, sPerm);
+
+			// system dialog asking for permission
+
+			// this code would not compile here - but it compile in your own app.
+			// to be compiled inside Uno, it has to be splitted into layers
+			//	var tcs = new TaskCompletionSource<Uno.UI.BaseActivity.RequestPermissionsResultWithResultsEventArgs>();
+
+			//	void handler(object sender, Uno.UI.BaseActivity.RequestPermissionsResultWithResultsEventArgs e)
+			//	{
+
+			//		if (e.RequestCode == 1)
+			//		{
+			//			tcs.TrySetResult(e);
+			//		}
+			//	}
+
+			//	var current = Uno.UI.BaseActivity.Current;
+
+			//	try
+			//	{
+			//		current.RequestPermissionsResultWithResults += handler;
+
+			//		ActivityCompat.RequestPermissions(Uno.UI.BaseActivity.Current, requestPermission.ToArray(), 1);
+
+			//		var result = await tcs.Task;
+			//		if (result.GrantResults.Length < 1)
+			//			return null;
+			//              foreach(var oItem in result.GrantResults)
+			//              {
+			//                  if (oItem == Android.Content.PM.Permission.Denied)
+			//                      return null;
+			//              }
+			//		return _histStore;
+
+			//	}
+			//	finally
+			//	{
+			//		current.RequestPermissionsResultWithResults -= handler;
+			//	}
+
+
+			//	return Windows.Devices.Geolocation.GeolocationAccessStatus.Denied;
+
+			//}

```

Comment:
This needs to be in a documentation file, you can take a look in the features directory.

Reply:
info in `//` was outdated (some PRs ago).